Plugins/FormatSwift/Plugin.swift

```diff
@@ -166,3 +167,92 @@ enum CommandError: Error {
   case lintFailure
   case unknownError(exitCode: Int32)
 }
+
+// MARK: - SwiftVersion
+
+struct SwiftVersion: Comparable {
+  var major: Int
+  var minor: Int
+
+  static func ==(_ lhs: SwiftVersion, _ rhs: SwiftVersion) -> Bool {
+    lhs.major == rhs.major
+      && lhs.minor == rhs.minor
+  }
+
+  static func <(_ lhs: SwiftVersion, _ rhs: SwiftVersion) -> Bool {
+    if lhs.major == rhs.major {
+      return lhs.minor < rhs.minor
+    } else {
+      return lhs.major < rhs.major
+    }
+  }
+}
+
+extension Package {
+
+  // MARK: Internal
+
+  /// The minimum Swift version supported by this package
+  var minimumSwiftVersion: SwiftVersion {
```

Comment:
How feasible would it be to add a couple of unit tests for this method?

Reply:
We don't currently have any unit tests for this package. What we do have is a CI job that builds and runs the package plugin, so verifies that any functionality exercised in the command builds and runs successfully.

In our case, as of https://github.com/airbnb/swift/pull/236 CI will fail if this doesn't work correctly. That provides a minimum guarantee that this works as expected.

Unit testing this logic would be a bit involved, especially since it works with the file system. We'd have to restructure the code to not directly interact with the filesystem, which adds complexity and overhead.

I find the existence of a CI job that fails without this to be a sufficient test case. Thoughts?

Reply:
I looked in to testing this logic, and there are some obstacles:

First, it isn't possible to unit test logic in a Swift package plugin. There is confirmation about this here: https://forums.swift.org/t/how-can-we-test-swiftpm-plugins/61279/2

After that didn't work I figured we can still extract the code we wanted to test into a new, separate module called "PluginHelpers" which can then be imported by the plugin. Unfortunately package plugins also can't even import other modules -- I get this error:

```
plugin 'FormatSwift' cannot depend on 'PluginHelpers' of type 'library'; this dependency is unsupported
```

If we want to test this logic we'd have to move it out of the plugin entirely and into the `AirbnbSwiftFormatTool` executable. That would add additional complexity to our communication between the plugin and executable, since we have to create a new way to represent all of the needed info in the arguments we use when invoking the executable.

Reply:
This is great investigation. Thank you @calda .

Reply:
Your findings have made me think that in general we should prefer to write code in the `AirbnbSwiftFormatTool` rather than the plugin. That being said, I see how it could complex to move this code into `AirbnbSwiftFormatTool` given context that only exists in the plugin. https://github.com/airbnb/swift/blob/8bd404ca686b72c1931d434f5286b6975833cc77/Plugins/FormatSwift/Plugin.swift#L210